RealmSwift/SwiftUI.swift

```diff
@@ -0,0 +1,619 @@
+    @available(iOS 14.0, macOS 11.0, tvOS 14.0, watchOS 7.0, *)
+    @StateObject private var storage: ObservableStorage<T>
+    private let defaultValue: T
+
+    /// :nodoc:
+    public var wrappedValue: T {
+        get {
+            if storage.value.realm == nil {
+                // if unmanaged return the unmanaged value
+                return storage.value
+            } else if storage.value.isInvalidated {
+                // if invalidated, return the default value
+                return defaultValue
+            }
+            // else return the frozen value. the frozen value
+            // will be consumed by SwiftUI, which requires
+            // the ability to cache and diff objects and collections
+            // during some timeframe. The ObjectType is frozen so that
+            // SwiftUI can cache state. other access points will thaw
+            // the ObjectType
+            return storage.value.freeze()
+        }
+        nonmutating set {
+            storage.value = newValue
+        }
+    }
+    /// :nodoc:
+    public var projectedValue: Binding<T> {
+        Binding(get: {
+            if self.storage.value.isInvalidated {
+                return self.defaultValue
+            }
+            return self.storage.value
+        }, set: { newValue in
+            self.storage.value = newValue
+        })
+    }
+    /**
+     Initialize a RealmState struct for a given thread confined type.
+     - parameter wrappedValue The List reference to wrap and observe.
+     */
+    @available(iOS 14.0, macOS 11.0, tvOS 14.0, watchOS 7.0, *)
+    public init<Value>(wrappedValue: T) where T == List<Value> {
+        self._storage = StateObject(wrappedValue: ObservableStorage(wrappedValue))
+        defaultValue = T()
+    }
+    /**
+     Initialize a RealmState struct for a given thread confined type.
+     - parameter wrappedValue The ObjectBase reference to wrap and observe.
+     */
+    @available(iOS 14.0, macOS 11.0, tvOS 14.0, watchOS 7.0, *)
+    public init(wrappedValue: T) where T: ObjectKeyIdentifiable {
+        self._storage = StateObject(wrappedValue: ObservableStorage(wrappedValue))
+        defaultValue = T()
+    }
+}
+
+// MARK: FetchRealmResults
+
+/// A property wrapper type that retrieves results from a Realm.
+///
+/// The results use the realm configuration provided by
+/// the environment value `EnvironmentValues/realmConfiguration`.
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.0, *)
+@propertyWrapper public struct FetchRealmResults<ResultType>: DynamicProperty, BoundCollection where ResultType: Object & ObjectKeyIdentifiable {
+    private class Storage: ObservableStorage<Results<ResultType>> {
+        var sortDescriptor: SortDescriptor? {
+            willSet {
+                if let sortDescriptor = newValue {
+                    value = baseValue.sorted(byKeyPath: sortDescriptor.keyPath, ascending: sortDescriptor.ascending)
```

Comment:
Rather than duplicating all this logic in a whole bunch of places there should just be a single function that updates things which is called in each didSet.